### PR TITLE
[master] Integrate Client generation for the Bal build command 

### PR DESCRIPTION
**Pull request by lnash94**
Fix: https://github.com/ballerina-platform/ballerina-library/issues/5822
Parent issue: https://github.com/ballerina-platform/ballerina-library/issues/4932

-This PR reduce merge commits from parent PR [1]
[1]Relevant PR https://github.com/ballerina-platform/openapi-tools/pull/1597

PRs: 
- https://github.com/ballerina-platform/openapi-tools/pull/1569
- https://github.com/ballerina-platform/openapi-tools/pull/1576
- https://github.com/ballerina-platform/openapi-tools/pull/1591
- https://github.com/ballerina-platform/openapi-tools/pull/1598

### CODE REVIEW

**Comment by lnash94** [comment]
PRs: 
- https://github.com/ballerina-platform/openapi-tools/pull/1569
- https://github.com/ballerina-platform/openapi-tools/pull/1576
- https://github.com/ballerina-platform/openapi-tools/pull/1591
- https://github.com/ballerina-platform/openapi-tools/pull/1598

**Review by dilanSachi — APPROVED** [review]

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ConstraintAnnotation.java
Comment (TharmiganK) [inline openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ConstraintAnnotation.java:1]: We can remove these files. With the restructuring the entire `openapi-bal-service` folder is removed

File: openapi-cli/src/main/java/io/ballerina/openapi/cmd/Add.java [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+            if (optionsBuilder.getTags() != null && !optionsBuilder.getTags().isEmpty()) {
+                moduleMembers = addNewLine(moduleMembers, 1);
+            }
+            Node[] arrayItems = getArrayItemNodes(getOperations());
+            SeparatedNodeList<ValueNode> value = createSeparatedNodeList(arrayItems);
+            ArrayNode arrayNode = NodeFactory.createArrayNode(createToken(SyntaxKind.OPEN_BRACKET_TOKEN), value,
+                    createToken(SyntaxKind.CLOSE_BRACKET_TOKEN));
+            KeyValueNode tagNodes = NodeFactory.createKeyValueNode(getKeyNode("options.operations"),
+                    getAssignToken(), arrayNode);
+            moduleMembers = moduleMembers.add(tagNodes);;
```
Comment (TharmiganK) [inline openapi-cli/src/main/java/io/ballerina/openapi/cmd/Add.java:236]:
```suggestion
            moduleMembers = moduleMembers.add(tagNodes);
```

File: openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+        List<TypeDefinitionNode> typeDefinitionNodeList = new LinkedList<>();
+        typeDefinitionNodeList.addAll(ballerinaClientGenerator.getTypeDefinitionNodeList());
+        typeDefinitionNodeList.addAll(ballerinaClientGenerator
+                .getBallerinaAuthConfigGenerator().getAuthRelatedTypeDefinitionNodes());
+        BallerinaTypesGenerator ballerinaSchemaGenerator = new BallerinaTypesGenerator(oasClientConfig.getOpenAPI(),
+                oasClientConfig.isNullable(), typeDefinitionNodeList);
+        SyntaxTree schemaSyntaxTree = ballerinaSchemaGenerator.generateSyntaxTree();
+        String schemaContent = Formatter.format(schemaSyntaxTree).toString();
+
+        if (oasClientConfig.getFilters().getTags().size() > 0) {
```
Comment (TharmiganK) [inline openapi-bal-task-plugin/src/main/java/io/ballerina/openapi/bal/tool/OpenAPICodeGeneratorTool.java:356]:
```suggestion
        if (oasClientConfig.getFilters().getTags().isEmpty()) {
```

File: openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ConstraintAnnotation.java
Comment (TharmiganK) [reply openapi-bal-service/src/main/java/io/ballerina/openapi/converter/service/ConstraintAnnotation.java:1]: Yes, we can delete the entire directory: `openapi-bal-service`

**Review by TharmiganK — APPROVED** [review]
LGTM

**Pull request merged**